apps/client/src/AppRouter.tsx

```diff
@@ -34,20 +35,19 @@ const Info = lazy(() => import('./features/info/InfoExport'));
 
 export default function AppRouter() {
   const { data } = useAliases();
-  const location = useLocation();
+  const [searchParams] = useSearchParams();
   const navigate = useNavigate();
+  const location = useLocation();
 
   // navigate if is alias route
   useEffect(() => {
     if (!data) return;
-
-    for (const d of data) {
-      if (`/${d.alias}` === location.pathname && d.enabled) {
-        navigate(`/${d.pathAndParams}`);
-        break;
-      }
+    const url = getAliasRoute(location, data, searchParams);
```

Comment:
I usually prefer returning something semantic like null.
This is personal I guess, and probably related to my C++ background, up to you